src/Drivers/StorageDriver.php

```diff
@@ -10,34 +10,41 @@
 class StorageDriver extends BaseDriver
 {
     protected Filesystem $disk;
-    protected ?string $prodFilePath;
+    protected ?string $mainFilePath;
+    protected array $config;
 
     public function __construct()
     {
         parent::__construct();
 
-        $this->disk = Storage::disk(config('auto-doc.drivers.storage.disk'));
-        $this->prodFilePath = config('auto-doc.drivers.storage.production_path');
+        $this->config = config('auto-doc.drivers.storage');
+        $this->disk = Storage::disk($this->config['disk']);
 
-        if (empty($this->prodFilePath)) {
+        $directory = str_ends_with($this->config['directory'], DIRECTORY_SEPARATOR)
```

Comment:
as you already did the same in the local driver, let's move this logic to the BaseDriver class

```suggestion
        $directory = (str_ends_with($this->config['directory'], DIRECTORY_SEPARATOR))
```